refactor(platform-browser): remove deprecated `BrowserModule.withServerTransition` method

alan-agius4 opened the pull request:
BREAKING CHANGE: The deprecated `BrowserModule.withServerTransition` method has been removed. Please use the `APP_ID` DI token to set the application id instead.

alan-agius4 commented:
Looks like there are still a couple of usages in G3 which require some cleanup.

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

alan-agius4 committed:
refactor(platform-browser): remove deprecated `BrowserModule.withServerTransition` method

BREAKING CHANGE: The deprecated `BrowserModule.withServerTransition` method has been removed. Please use the `APP_ID` DI token to set the application id instead.

AndrewKushnir commented:
This PR was merged into the repository by commit 5c61f46409855bb8fe66d71a9c16c00753032987.

The changes were merged into the following branches: main
